warehouse/manage/views.py

```diff
@@ -1157,6 +1162,9 @@ def manage_project_oidc_providers(self):
 
     @view_config(request_method="POST", request_param=GitHubProviderForm.__params__)
     def add_github_oidc_provider(self):
+        if not self.oidc_enabled:
+            return self.default_response
```

Comment:
I think this should 404 instead.

Reply:
[c2ca980](https://github.com/pypa/warehouse/pull/10753/commits/c2ca980fc4ae3551631d5434212b4b7a0d2f1085)